Realm/RLMLogger.mm

```diff
@@ -60,90 +58,203 @@ static RLMLogLevel logLevelForLevel(Level logLevel) {
         case RLMLogCategorySyncClientChangeset: return LogCategory::changeset;
-        case RLMLogCategorySyncClientNetwork: return LogCategory::network;
-        case RLMLogCategorySyncClientReset: return LogCategory::reset;
-        case RLMLogCategorySyncServer: return LogCategory::server;
+        case RLMLogCategorySyncClientNetwork:   return LogCategory::network;
+        case RLMLogCategorySyncClientReset:     return LogCategory::reset;
+        case RLMLogCategorySyncServer:          return LogCategory::server;
     };
     REALM_UNREACHABLE();
 }
 
-static RLMLogCategory logCategoryForCategoryName(std::string category) {
-    NSDictionary *categories = @{
-        @"Realm": @(RLMLogCategoryRealm),
-        @"Realm.SDK": @(RLMLogCategorySDK),
-        @"Realm.App": @(RLMLogCategoryApp),
-        @"Realm.Storage": @(RLMLogCategoryStorage),
-        @"Realm.Storage.Transaction": @(RLMLogCategoryStorageTransaction),
-        @"Realm.Storage.Query": @(RLMLogCategoryStorageQuery),
-        @"Realm.Storage.Object": @(RLMLogCategoryStorageObject),
-        @"Realm.Storage.Notification": @(RLMLogCategoryStorageNotification),
-        @"Realm.Sync": @(RLMLogCategorySync),
-        @"Realm.Sync.Client": @(RLMLogCategorySyncClient),
-        @"Realm.Sync.Client.Session": @(RLMLogCategorySyncClientSession),
-        @"Realm.Sync.Client.Changeset": @(RLMLogCategorySyncClientChangeset),
-        @"Realm.Sync.Client.Network": @(RLMLogCategorySyncClientNetwork),
-        @"Realm.Sync.Client.Reset": @(RLMLogCategorySyncClientReset),
-        @"Realm.Sync.Server": @(RLMLogCategorySyncServer)
+static RLMLogCategory logCategoryForCategory(const LogCategory& category) {
+    static constinit std::pair<const LogCategory*, RLMLogCategory> categories[] = {
+        {&LogCategory::realm, RLMLogCategoryRealm},
+        {&LogCategory::sdk, RLMLogCategorySDK},
+        {&LogCategory::app, RLMLogCategoryApp},
+        {&LogCategory::storage, RLMLogCategoryStorage},
+        {&LogCategory::transaction, RLMLogCategoryStorageTransaction},
+        {&LogCategory::query, RLMLogCategoryStorageQuery},
+        {&LogCategory::object, RLMLogCategoryStorageObject},
+        {&LogCategory::notification, RLMLogCategoryStorageNotification},
+        {&LogCategory::sync, RLMLogCategorySync},
+        {&LogCategory::client, RLMLogCategorySyncClient},
+        {&LogCategory::session, RLMLogCategorySyncClientSession},
+        {&LogCategory::changeset, RLMLogCategorySyncClientChangeset},
+        {&LogCategory::network, RLMLogCategorySyncClientNetwork},
+        {&LogCategory::reset, RLMLogCategorySyncClientReset},
+        {&LogCategory::server, RLMLogCategorySyncServer},
     };
-    if (NSNumber *logCategory = [categories objectForKey:RLMStringDataToNSString(category)]) {
-        return RLMLogCategory([logCategory intValue]);
-    }
-    REALM_UNREACHABLE();
+
+    auto find = [](const LogCategory& category) {
+        for (auto& [core, objc] : categories) {
```

Comment:
The lookup table is a compile-time constant rather than a hash table constructed at runtime, and constructing that hash table takes a lot longer than the actual lookup.